yarn-project/prover-node/src/prover-coordination/factory.ts

```diff
@@ -46,27 +61,31 @@ export async function createProverCoordination(
     if (!deps.archiver || !deps.worldStateSynchronizer || !deps.telemetry || !deps.epochCache) {
       throw new Error('Missing dependencies for p2p prover coordination');
     }
-
-    const proofVerifier = config.realProofs ? await BBCircuitVerifier.new(config) : new TestCircuitVerifier();
-    const p2pClient = await createP2PClient(
-      P2PClientType.Prover,
-      config,
-      deps.archiver,
-      proofVerifier,
-      deps.worldStateSynchronizer,
-      deps.epochCache,
-      deps.telemetry,
-    );
-    await p2pClient.start();
-
-    return p2pClient;
   }
 
-  if (config.proverCoordinationNodeUrl) {
-    log.info('Using prover coordination via node url');
+  let nodes: AztecNode[] = [];
+  if (config.proverCoordinationNodeUrls.length > 0) {
+    log.info('Using prover coordination via node urls');
     const versions = getComponentsVersionsFromConfig(config, protocolContractTreeRoot, getVKTreeRoot());
-    return createAztecNodeClient(config.proverCoordinationNodeUrl, versions, makeTracedFetch([1, 2, 3], false));
-  } else {
-    throw new Error(`Aztec Node URL for Tx Provider is not set.`);
+    nodes = config.proverCoordinationNodeUrls.map(url =>
+      createAztecNodeClient(url, versions, makeTracedFetch([1, 2, 3], false)),
+    );
   }
+
+  const proofVerifier = config.realProofs ? await BBCircuitVerifier.new(config) : new TestCircuitVerifier();
+  const p2pClient = await createP2PClient(
+    P2PClientType.Prover,
+    config,
+    deps.archiver,
+    proofVerifier,
+    deps.worldStateSynchronizer,
+    deps.epochCache,
+    deps.telemetry,
+  );
+  await p2pClient.start();
+
+  return {
+    proverCoordination: new CombinedProverCoordination(p2pClient, nodes, coordinationConfig),
+    p2pClient: p2pClient,
+  };
```

Comment:
Nit: I'd add a `getP2PClient(): P2PClient | undefined` method to the ProverCoordination API, so we can keep returning just the `proverCoordination` here. But it's just a cosmetic change.

Reply:
Done